example/osp_deletion_example.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (C) 2018 Red Hat, Inc.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+import ast
+from miqcli import Client
+from time import sleep
+
+# The input to deletion is the name of the vm
+INPUT = "<vm_name_to_delete>"
+
+# create a client object
+# pass in config, if empty {} means it will use the default credentials
+config = {
+    'username': 'admin',
+    'password': 'smartvm',
+    'url': 'https://localhost:8443',
+    'enable_ssl_verify': False
+}
+
+client = Client(config)
+fip_id = None
+
+# 1. Query the instance to delete to see if the instance has a floating ip
+# set the id of the floating ip as fip_id
+client.collection = "instances"
+try:
+    instances = client.collection.query(inst_name=INPUT,
+                                        attr="floating_ip")
+    fip_id = instances["floating_ip"]["id"]
+except SystemExit as e:
+    print('No associated floating ips for instance: {0}, will continue to'
+          ' delete.'.format(INPUT))
+    instances = client.collection.query(inst_name=INPUT, attr=None)
+
+# 2. If there is a floating ip, detach it and remove it
+if fip_id:
+    payload_data = {'floating_ip_id': fip_id}
+    # detach the floating ip
+    client.collection = "automation_requests"
+    req_id = client.collection.create(
+        method='release_floating_ip',
+        payload=str(payload_data),
+        payload_file=None)
+    print(req_id)
+
+    # 3. the script will keep querying automate_request for the status to be
+    #  finished, once finished, it will query for the floating ip id. If there
+    #  was an error, the script will display the error message and exit.
+    done = False
+    while not done:
+        result = client.collection.status(req_id)
+        if result.request_state == "finished":
+            done = True
+        sleep(5)
+
+    # see the return and options of the automate request
+    print("Options from the automate request: {0}".format(result.options))
+
+    # verify from the automate output that a floating ip was successfully
+    # created, and extract the id of the floating ip address
```

Comment:
Comment says created. Should it not state deleted.

Reply:
yes, that needs to be changed.